src/document.c

```diff
@@ -573,6 +574,7 @@ int Document_AddToIndexes(RSAddDocumentCtx *aCtx) {
 
       PreprocessorFunc pp = preprocessorMap[ii];
       if (pp(aCtx, &doc->fields[i], fs, fdata, &aCtx->status) != 0) {
+        ++aCtx->stats->indexFailures;
```

Comment:
This is not safe when running in concurrent write mode, We need to see if we blocked and if we did we need to take the GIL before do it (and make sure the index still exists).